x-pack/test/functional/apps/canvas/index.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { EsArchiver } from '@kbn/es-archiver';
+import { FtrProviderContext } from '../../ftr_provider_context';
+export default ({ loadTestFile, getService }: FtrProviderContext) => {
+  const security = getService('security');
+  const config = getService('config');
+  let esNode: EsArchiver;
+
+  describe('Canvas', function canvasAppTestSuite() {
+    describe('Canvas app', () => {
+      before(async () => {
+        // init data
+        const roles = [
+          'test_logstash_reader',
+          'global_canvas_all',
+          'global_discover_all',
+          'global_maps_all',
+          // TODO: Fix permission check, save and return button is disabled when dashboard is disabled
+          'global_dashboard_all',
+        ];
+        if (config.get('esTestCluster.ccs')) roles.push('ccs_remote_search');
+        await security.testUser.setRoles(roles);
+        esNode = config.get('esTestCluster.ccs')
+          ? getService('remoteEsArchiver' as 'esArchiver')
```

Comment:
You know I know I need to work on my Typescript game, but wow...I had no idea you could throw an `as` in here like this.  Maybe we can chat about how this actually works :)

Reply:
Let's zoom about it.